### PR TITLE
wallet-rpc: get balance for all accounts and subaddresses

### PR DESCRIPTION
**Pull request by stoffu**

### CODE REVIEW

File: src/wallet/wallet_rpc_server.cpp [hunk 1]
```diff
@@ -335,32 +335,62 @@ namespace tools
   bool wallet_rpc_server::on_getbalance(const wallet_rpc::COMMAND_RPC_GET_BALANCE::request& req, wallet_rpc::COMMAND_RPC_GET_BALANCE::response& res, epee::json_rpc::error& er, const connection_context *ctx)
   {
     if (!m_wallet) return not_open(er);
+    if (req.all_accounts && !req.address_indices.empty())
```
Comment (jtgrassie) [inline src/wallet/wallet_rpc_server.cpp]: I wouldn't bother with this check. Just have all_accounts override.

Comment (stoffu) [reply src/wallet/wallet_rpc_server.cpp]: OK, updated

File: src/wallet/wallet_rpc_server.cpp [hunk 2]
```diff
@@ -337,30 +337,54 @@ namespace tools
...
+          wallet_rpc::COMMAND_RPC_GET_BALANCE::per_subaddress_info info;
+          info.account_index = account_index;
+          info.address_index = i;
+          cryptonote::subaddress_index index = {info.account_index, info.address_index};
+          info.address = m_wallet->get_subaddress_as_str(index);
+          info.balance = balance_per_subaddress[i];
+          info.unlocked_balance = unlocked_balance_per_subaddress[i];
+          info.label = m_wallet->get_subaddress_label(index);
+          info.num_unspent_outputs = std::count_if(transfers.begin(), transfers.end(), [&](const tools::wallet2::transfer_details& td) { return !td.m_spent && td.m_subaddr_index == index; });
+          res.per_subaddress.push_back(info);
```
Comment (mudlotus) [inline src/wallet/wallet_rpc_server.cpp]:
```suggestion
          res.per_subaddress.emplace_back(std::move(info));
```

Comment (stoffu) [reply src/wallet/wallet_rpc_server.cpp]: Thanks, updated

**Pull request merged**